packages/techdocs-common/src/stages/publish/awsS3.test.ts

```diff
@@ -127,11 +135,15 @@ describe('AwsS3Publish', () => {
         apiVersion: 'apiVersion',
         kind: 'kind',
         metadata: {
-          namespace: '/namespace',
+          namespace: 'namespace',
           name: 'name',
         },
       };
-      const entityRootDir = `${entityMock.metadata.namespace}/${entityMock.kind}/${entityMock.metadata.name}`;
+      const entityRootDir = path.join(
+        entityMock.metadata.namespace,
+        entityMock.kind,
+        entityMock.metadata.name,
```

Comment:
Pardon my eyes here, but this join is entityMock with two (2) of three (3) elements under `metadata.`, with `kind` just being off the root object.  The other two places where the path is created (lines 175, 201), are just straight off `entityMock` with no metadata subselector.

Is that intentional?  Only noting it cause it looked funny is all but it may be perfectly legit for this test!! 😁

Reply:
Good observation! 😄 

This `entityMock` here is created using the `createMockEntity` function defined above on top of the file, which creates an entity object very similar to the real thing https://github.com/backstage/backstage/blob/master/packages/catalog-model/src/entity/Entity.ts

The `entityMock` below (line 173) seems to have a different custom structure than how an entity looks. (Should be changed IMO). :)

Reply:
Thanks for the observation ! I'll fix that 👍

Reply:
I did a little bit of refactoring to make the tests easier to read.
I added an helper `createMockEntityName` which return an `EntityName` to pass to the `fetchTechDocsMetadata` method.

```ts
const createMockEntityName = (): EntityName => ({
  kind: 'TestKind',
  name: 'test-component-name',
  namespace: 'test-namespace',
});
```

I think it will be easier to understand by defining the return types of the mock creation functions 👍

Reply:
Aah, so one was `Entity` and the other was `EntityName`. That has been my "Today I Learned" for many days now.
Thanks for clarifying with the refactor @ayshiff !